accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,126 @@
+
+The compiler has a very stable 1:1 relationship between a user's source and the code it generates. As such the user has the ability to control the output of the compiler through their own usage.  The compiler's contribution of complexity to the user's assembly is very small and highly specified.
+ 
+Contrast this to source generators which can take a very small amount of user source to produce a large amount of generated code.  This large amount of generated code contains significant complexity which is loosely specified.  It's behavior is specific to the component it extends.  It's likely that we'd need to bugfix this code over time, and change it as we develop features in the components it extends.  It's also likely that we will learn new techniques and improvments to source generators over time, increasing the changes for bugfixes and features
+
+User's expect stability for projects which do not change, for example a project targeting an LTS release. It is much more difficult to ensure this stability when we need to significantly components which contribute complexity to that project.
+
+### Is it bad to have multiple versions of a source generator possibly loaded by a repository?
+
+A repository may have projects targeting multiple frameworks.  These frameworks may contain different versions of the same source generator.  When building the repository this may require the same compiler server instance to load multiple copies of a source generator.  The same compiler would not likely load multiple copies of analyzers which do not ship side by side.  
+
+### Should all inbox source generators be tied to a shared framework?
+
+No, some source generators might be independent of libraries in a shared framework: these could go in a non-side-by-side location (eg: analyzers folder).  Other source generators may not contain functionality, but may instead just build some convenience API on top of existing public API.  These could be considered stable enough to place in a non-side-by-side location.
+
+### Should only source generators be shipped this way?  What about analyzers and code-fixes?
+
+Nothing is preventing analyzers and code fixes from shipping in the same manner.  If it's desireable for a such a component to ship with the framework rather than the compiler then that component may ship in this manner.
+
+### Why in the ref pack, what not some other "analyzer" pack?
+
+The SDK already has a means for acquisition and selection of ref packs.  The inputs to the selection of ref packs are the same as that of source generators (Version, FrameworkReference, and TargetFramework).  Ref packs user normal nuget convention which support analyzers and source generators.
+
+It may be desireable to create a seperate package just for analyzers in order to reduce risk in servicing.  We support servicing ref packs today, but we have to be careful not to expose new API nor change any assembly identites.  
```

Comment:
```suggestion
It may be desirable to create a separate package just for analyzers in order to reduce risk in servicing.  We support servicing ref packs today, but we have to be careful not to expose new API nor change any assembly identities.  
```